### DISCUSSION TITLE
Regelstrategie 2404

### DISCUSSION THREAD

**Discussion by cvbasta75** [open3e/open3e#131] [Q&A]
Die Regelstrategie lässt sich aber leider nicht schreiben. Vielleicht weiß einer mehr. In der ViCare App und an der Heizung an sich kann man es ja auch ändern...

Hier ändert sich zwar immer nur die letzte Zahl in den Raw-Daten aber:

`/root/pp/bin/open3e -c can0 -r 2404 --raw -v
None 2404 BivalenceControlMode "029600000002"
closing 0x680 - bye!

/root/pp/bin/open3e -c can0 -w 2404=029600000001 --raw -v
write raw: 1664.2404 = 029600000001
Traceback (most recent call last):
File "/root/pp/bin/open3e", line 8, in
sys.exit(main())
^^^^^^
File "/root/pp/lib/python3.12/site-packages/open3e/Open3Eclient.py", line 386, in main
succ,code = dicEcus[ecu].writeByDid(didkey, didVal, raw=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/root/pp/lib/python3.12/site-packages/open3e/Open3Eclass.py", line 162, in writeByDid
response = self.uds_client.write_data_by_identifier(did, val)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/root/pp/lib/python3.12/site-packages/udsoncan/client.py", line 167, in decorated
return func(self, *args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/root/pp/lib/python3.12/site-packages/udsoncan/client.py", line 521, in write_data_by_identifier
response = self.send_request(req)
^^^^^^^^^^^^^^^^^^^^^^
File "/root/pp/lib/python3.12/site-packages/udsoncan/client.py", line 2258, in send_request
raise NegativeResponseException(response)
udsoncan.exceptions.NegativeResponseException: WriteDataByIdentifier service execution returned a negative response ConditionsNotCorrect (0x22)`

**Comment by philippoo66** [comment]
moin  @cvbasta75 !

versuch's mal damit: https://github.com/open3e/open3e/tree/service77

Grüsse!
Phil

**Reply by MyHomeMyData** [reply]
@cvbasta75 Der SW-Stand hat noch eine geringe Test-Tiefe! Generell funktioniert der Service 77 nur auf dem **internen CAN-Bus**.

Es sollte aber möglich sein, damit den Datenpunkt 2404 zu ändern.

**Reply by MyHomeMyData** [reply]
@cvbasta75 Im Branch develop ist jetzt eine gut getestete Version verfügbar, die Service 77 beherrscht. Damit solltest Du 2404 ändern können:
* CAN-Adapter mit den **internen CAN-Bus** verbinden. Nach unseren bisherigen Erfahrungen funktioniert open3e auf dem internen Bus genauso gut wie auf externen.
* Um Service 77 anzufordern, die Option `-f77` verwenden.
